app/components/app/app.service.js

```diff
@@ -0,0 +1,29 @@
+class AppService {
+    constructor($http, BASE_URL) {
+        'ngInject';
+
+        this.$http = $http;
+        this.BASE_URL = BASE_URL;
+    }
+
+    getCategories() {
+        return this.$http.get(`${this.BASE_URL}categories.json`).then((response) => {
+            return response.data;
+        })
+        .catch((response) => {
+            console.log(response.statusText);
+        });
+    }
+
+    getProducts() {
+        return this.$http.get(`${this.BASE_URL}products.json`).then((response) => {
```

Comment:
One more thing - delete the final slash in BASE_URL, so links here will look much better with slash before `products.json`